fineract-provider/src/main/java/org/apache/fineract/infrastructure/campaigns/sms/service/SmsCampaignDropdownReadPlatformServiceImpl.java

```diff
@@ -70,19 +71,18 @@ public SmsCampaignDropdownReadPlatformServiceImpl(final SmsConfigUtils smsConfig
     public Collection<SmsProviderData> retrieveSmsProviders() {
         Collection<SmsProviderData> smsProviderOptions = new ArrayList<>();
         String hostName = "" ;
-        try {
             Map<String, Object> hostConfig = this.smsConfigUtils.getMessageGateWayRequestURI("smsbridges", null);
             URI uri = (URI) hostConfig.get("uri");
             hostName = uri.getHost() ;
             HttpEntity<?> entity = (HttpEntity<?>) hostConfig.get("entity");
             ResponseEntity<Collection<SmsProviderData>> responseOne = restTemplate.exchange(uri, HttpMethod.GET, entity,
                     new ParameterizedTypeReference<Collection<SmsProviderData>>() {});
+             if (!responseOne.getStatusCode().equals(HttpStatus.OK)) {
```

Comment:
I'm not 100% sure if PlatformDataIntegrityException is "correct" here, but don't really have a better idea without more closely than I have time for, and it probably doesn't really matter all that much, anyway. One fine day we should look into and then write up Guidance in the README about which exact Fineract Exception type to use in which "layer" (e.g. *Resource VS *Service). @ptuomola I doubt that's something you would fancy digging into?